integrations/whatsapp/src/channels/message-types/card.ts

```diff
@@ -0,0 +1,156 @@
+import { Text, Interactive, ActionButtons, Header, Image, Button, ActionCTA } from 'whatsapp-api-js/messages'
+import { WHATSAPP } from '../../misc/constants'
+import { chunkArray, hasAtleastOne } from '../../misc/util'
+import * as body from './interactive/body'
+import * as button from './interactive/button'
+import * as footer from './interactive/footer'
+import { channels } from '.botpress'
+import * as bp from '.botpress'
+
+type Card = channels.whatsapp.card.Card
+
+type SDKAction = Card['actions'][number]
+type ActionURL = SDKAction & { action: 'url' }
+type ActionSay = SDKAction & { action: 'say' }
+type ActionPostback = SDKAction & { action: 'postback' }
+
+type Action = ActionSay | ActionURL | ActionPostback
+
+type ActionsChunk = ReturnType<typeof chunkArray<ActionSay | ActionPostback>>[number]
+
+export function* generateOutgoingMessages(card: Card, logger: bp.Logger) {
```

Comment:
~~**question:** why a generator ?~~

**edit:** ok, got it: your send each generated message in between iterations